From stable-bounces@linux.kernel.org  Wed May 23 15:44:49 2007
Message-ID: <4654C38C.2010804@redhat.com>
Date: Wed, 23 May 2007 18:43:24 -0400
From: Chuck Ebbert <cebbert@redhat.com>
To: stable@kernel.org
Cc: "Brown, Len" <len.brown@intel.com>
Subject: acpi-thermal: fix mod_timer() interval

From:  Andrew Morton <akpm@linux-foundation.org>

Use relative time, not absolute.  Discovered by Jung-Ik (John) Lee
<jilee@google.com>.

Cc: Jung-Ik (John) Lee <jilee@google.com>
Acked-by: Len Brown <lenb@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/acpi/thermal.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- linux-2.6.20.13.orig/drivers/acpi/thermal.c
+++ linux-2.6.20.13/drivers/acpi/thermal.c
@@ -758,7 +758,8 @@ static void acpi_thermal_check(void *dat
 			del_timer(&(tz->timer));
 	} else {
 		if (timer_pending(&(tz->timer)))
-			mod_timer(&(tz->timer), (HZ * sleep_time) / 1000);
+			mod_timer(&(tz->timer),
+					jiffies + (HZ * sleep_time) / 1000);
 		else {
 			tz->timer.data = (unsigned long)tz;
 			tz->timer.function = acpi_thermal_run;
